app/models/post.rb

```diff
@@ -1,8 +1,9 @@
 class Post < ApplicationRecord
   after_save :update_post_counter
+  after_destroy :decrement_post_counter
   belongs_to :author, class_name: 'User', foreign_key: 'author_id'
```

Comment:
- [OPTIONAL] You can add this line in `Post Controller` as you did with the `Comment Controller` to make sure to use **CanCanCan** for Post authorization

   ```
     class PostsController < ApplicationController
       load_and_authorize_resource
   
       ...
     end
   ```